Framework/PythonInterface/plugins/algorithms/SaveReflections.py

```diff
@@ -466,7 +491,7 @@ def __call__(self, file_name, workspace, _, scale):
 
         from mantid.simpleapi import SaveHKL
 
-        SaveHKL(Filename=file_name, InputWorkspace=workspace, OutputWorkspace=workspace.name(), ScalePeaks=scale)
+        SaveHKL(Filename=file_name, InputWorkspace=workspace, ScalePeaks=scale)
```

Comment:
I think the error is actually because `SaveHKL` doesn't work for `LeanElasticPeaks` - it currently calls methods such as `getDetectorID` and `getBankName` so it's not trivial to support them either.

So for example this works
```
LoadEmptyInstrument(InstrumentName='SXD', OutputWorkspace='SXD')
peaks = CreatePeaksWorkspace(InstrumentWorkspace='SXD', NumberOfPeaks=2, OutputWorkspace='peaks')
SaveHKL(Filename=r"C:\Users\xhg73778\Desktop\test2.hkl", InputWorkspace=peaks, OutputWorkspace="test", ScalePeaks=1, StoreInADS=False)
```
Changing to `OutputType='LeanElasticPeak'` produces this error
```
ValueError: Problem setting "InputWorkspace" in SaveHKL-v1: Invalid value for property InputWorkspace (PeaksWorkspace) from string "peaks": Workspace peaks is not of the correct type
```

Reply:
I agree, it would be better to not refactor `SaveHKL`. One option is to use `SaveHKLCW` which works with the LeanPeak workspace. It provides a constant wavelength format with or without direction cosines that similar to `SaveHKL`

Reply:
This would only be used if the provided workspace is a `LeanWorkspace`

Reply:
Good idea, it says in the docs `SaveHKLCW` saves to `SHELX76` format do you know if GSAS (or GSAS-II) can read this format (I don't use GSAS and a quick google doesn't throw up much...). Probably worth checking GSAS can read what `SaveHKL` produces...

Reply:
I have one GSAS file. It looks like it will work, but I will still check

```
#Single crystal data (hb3a)#
   -0    1   -0  244.02   13.24
   -1   -0   -0 1009.99   57.60
    1    0   -0 1006.03   54.50
   -0    1   -1  856.64   46.40
 ```